src/VisualStudio/Core/Def/Implementation/ProjectSystem/DocumentProvider.StandardTextDocument.cs

```diff
@@ -49,7 +49,7 @@ private class StandardTextDocument : ForegroundThreadAffinitizedObject, IVisualS
                 DocumentProvider documentProvider,
                 IVisualStudioHostProject project,
                 DocumentKey documentKey,
-                IReadOnlyList<string> folderNames,
+                Func<uint, IReadOnlyList<string>> getFolderNames,
```

Comment:
Done - it's a VSITEMID.  Lambda is required so that we can defer getting the itemid to later.